File: test/resources/repos/cpp/test_repo/ChildClass.cpp
#include "ChildClass.h"
#include <iostream>

ChildClass::ChildClass(const std::string& name, int value) 
    : BaseClass(name), value(value) {}

void ChildClass::process() {
    std::cout << "Processing " << name << " with value " << value << std::endl;
}

int ChildClass::getValue() const {
    return value;
}

File: test/resources/repos/typescript/test_repo/ChildClass.ts
import { BaseClass } from './BaseClass';

export class ChildClass extends BaseClass {
    private value: number;

    constructor(name: string, value: number) {
        super(name);
        this.value = value;
    }

    execute(): void {
        console.log(`Executing ${this.name} with value ${this.value}`);
    }

    getValue(): number {
        return this.value;
    }
}